### PR TITLE
Refactor common_test_utils

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by olpipi** [comment]
@iefode @ilyachur @vurusovs 
Folks, could you please take a look at this PR?

File: src/core/tests/rtti.cpp [hunk 1]
```diff
@@ -2,9 +2,9 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "common_test_utils/all_close_f.hpp"
 #include "common_test_utils/test_tools.hpp"
 #include "gtest/gtest.h"
+#include "ngraph/function.hpp"
```
Comment (vurusovs) [inline src/core/tests/rtti.cpp:7]: @olpipi will the file be refactored and when?

File: src/tests/test_utils/common_test_utils/include/common_test_utils/all_close.hpp [hunk 2]
```diff
@@ -128,10 +85,15 @@ namespace test {
 /// Returns true if shapes match and for all elements, |a_i-b_i| <= atol + rtol*|b_i|.
 template <typename T>
 ::testing::AssertionResult all_close(const ov::Tensor& a, const ov::Tensor& b, T rtol = 1e-5f, T atol = 1e-8f) {
-    auto a_t = std::make_shared<ngraph::runtime::HostTensor>(a.get_element_type(), a.get_shape(), a.data());
-    auto b_t = std::make_shared<ngraph::runtime::HostTensor>(b.get_element_type(), b.get_shape(), b.data());
+    std::vector<T> a_v(a.get_size());
+    ov::Tensor a_t(a.get_element_type(), a.get_shape(), static_cast<void *>(a_v.data()));
+    a.copy_to(a_t);
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/all_close.hpp:90]: Is it possible not to copy memory? Looks like vector may be created over `a.data()`

File: src/tests/test_utils/common_test_utils/src/all_close_f.cpp [hunk 3]
```diff
@@ -4,24 +4,14 @@
...
-union FloatUnion {
-    float f;
-    uint32_t i;
-};
 
-union DoubleUnion {
-    double d;
-    uint64_t i;
-};
+namespace ngraph {
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/src/all_close_f.cpp:13]: should it be renamed?

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/core/tests/rtti.cpp (hunk 1 above)
Comment (ilyachur) [reply src/core/tests/rtti.cpp:7]: Here we can use new namespaces

File: src/tests/test_utils/common_test_utils/include/common_test_utils/all_close.hpp (hunk 2 above)
Comment (ilyachur) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/all_close.hpp:90]: If shapes are the same we can avoid copy

File: src/core/tests/rtti.cpp (hunk 1 above)
Comment (olpipi) [reply src/core/tests/rtti.cpp:7]: Yes, I change it

**Review by iefode — COMMENTED** [review]
@olpipi Please fix the comments. In other LGTM

**Review by vurusovs — APPROVED** [review]

**Pull request merged**